helix-core/src/main/java/org/apache/helix/HelixAdmin.java

```diff
@@ -280,6 +281,16 @@ void addResource(String clusterName, String resourceName, int numPartitions, Str
    */
   void enableInstance(String clusterName, String instanceName, boolean enabled);
 
+  /**
+   * @param clusterName
+   * @param instanceName
+   * @param enabled
+   * @param reason set additional string description on why the instance is disabled when
+   *          <code>enabled</code> is false.
+   */
+  void enableInstance(String clusterName, String instanceName, boolean enabled,
+      InstanceConstants.InstanceDisabledType disabledType, String reason);
+
```

Comment:
A general question, in such case of adding new method to interface, should we provide a default method impl for compatibility?

Reply:
Thanks for the review. Could you please elaborate a bit on default method impl?

Reply:
Yeah, that refers to default method of an interface, fyi https://www.baeldung.com/java-static-default-methods
This is to preserve backward compatibility, not sure how we usually do when adding or changing interface.

Reply:
I think for this particular change, since `HelixAdmin` is not a public interface that user depends on, we do not need to add default method impl. We probably need default method impl for public interface in `api` folder.